src/util.cpp

```diff
@@ -3602,6 +3602,35 @@ static std::unordered_map<std::string,int> g_usedNames;
 static std::mutex g_usedNamesMutex;
 static int g_usedNamesCount=1;
 
+/*! This function converts the value of CREATE_SUBDIRS
+ *  into the number of relevant bits for the number of
+ *  created directories in the 2nd level.
+ */
+static int convertCreateSubDirs2Level(CREATE_SUBDIRS_t createSubDirs)
+{
+  switch (createSubDirs)
+  {
+    case CREATE_SUBDIRS_t::D16:
+      return 0;
+    case CREATE_SUBDIRS_t::D32:
+      return 1;
+    case CREATE_SUBDIRS_t::D64:
+      return 2;
+    case CREATE_SUBDIRS_t::D128:
+      return 3;
+    case CREATE_SUBDIRS_t::D256:
+      return 4;
+    case CREATE_SUBDIRS_t::D512:
+      return 5;
+    case CREATE_SUBDIRS_t::D1024:
+      return 6;
+    case CREATE_SUBDIRS_t::D2048:
+      return 7;
+    default: /* NO, YES, D4096 */
+      return 8;
```

Comment:
I understand that you set here also in case of `NO` to 8 from a convenience point of view, so when entering by accident the array `createSubdirsBitmaskL2` one doesn't get out of bounds but seen the code `convertNameToFile` this cannot happen maybe a better way would be to have the value `-1` and place a warning in this routine that we should never get in the `NO` part by means of an `assert`.